sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OpenAiController.java

```diff
@@ -134,72 +114,47 @@ private static String objectToJson(@Nonnull final Object obj) {
     }
   }
 
-  /**
-   * Chat request to OpenAI with an image
-   *
-   * @return a ResponseEntity with the response content
-   */
   @GetMapping("/chatCompletionImage")
   @Nonnull
-  ResponseEntity<String> chatCompletionImage(
-      @Nullable @RequestHeader(value = "accept", required = false) final String accept)
-      throws JsonProcessingException {
+  Object chatCompletionImage(
+      @Nullable @RequestParam(value = "view", required = false) final String view) {
     final var response =
         service.chatCompletionImage(
             "https://upload.wikimedia.org/wikipedia/commons/thumb/5/59/SAP_2011_logo.svg/440px-SAP_2011_logo.svg.png");
-    if ("application/json".equals(accept)) {
-      return ResponseEntity.ok().body(MAPPER.writeValueAsString(response));
+    if ("json".equals(view)) {
+      return response;
     }
-    return ResponseEntity.ok(response.getContent());
+    return response.getContent();
   }
 
-  /**
-   * Chat request to OpenAI with a tool.
-   *
-   * @return a ResponseEntity with the response content
-   */
   @GetMapping("/chatCompletionTool")
   @Nonnull
-  ResponseEntity<String> chatCompletionTools(
-      @Nullable @RequestHeader(value = "accept", required = false) final String accept)
-      throws JsonProcessingException {
+  Object chatCompletionTools(
+      @Nullable @RequestParam(value = "view", required = false) final String view) {
     final var response =
         service.chatCompletionTools("Calculate the Fibonacci number for given sequence index.");
-    if ("application/json".equals(accept)) {
-      return ResponseEntity.ok().body(MAPPER.writeValueAsString(response));
+    if ("json".equals(view)) {
+      return response;
     }
-    return ResponseEntity.ok(response.getContent());
+    return response.getContent();
   }
 
-  /**
-   * Get the embedding of a text
-   *
-   * @return a ResponseEntity with the response content
```

Comment:
(Question)

Why remove javadoc? That's one way to boost positive diff rate ;)

Reply:
https://github.com/SAP/ai-sdk-java/pull/306#discussion_r1932211631